azurerm - error creating vm from existing vhd - osType not supplied

### Terraform Version
v0.6.15

### Affected Resource(s)
azurerm_virtual_machine

### Terraform Configuration Files
```
resource "azurerm_virtual_machine" "ap-service-discovery-server-0" {
    name = "ap-service-discovery-server-0"
    resource_group_name = "${var.resource_group_name}"
    location = "${var.location}"
    vm_size = "${var.instance_type}"
    network_interface_ids = ["${azurerm_network_interface.ap-service-discovery-server-0.id}"]
    availability_set_id = "${azurerm_availability_set.ap-service-discovery-availability-set.id}"

    #TODO at this point - copy from vhd storage, rename and move to VM storage acct.
    storage_os_disk {
        name = "ap-service-discovery-server-0-osdisk"
        vhd_uri = "${replace("${var.vhd_path}",".vhd","-0.vhd")}"
        create_option = "attach"
    }

    os_profile {
        computer_name = "ap-service-discovery-server-0"
        admin_username = "${var.ssh_username}"
        admin_password = "${var.ssh_password}"
        custom_data = "${template_file.user_data_0.rendered}"
    }

    os_profile_linux_config {
       disable_password_authentication = false
        /*ssh_keys {
          path = "/home/${var.ssh_username}/.ssh/authorized_keys"
          key_data = "${file("${var.ssh_key_path}")}"
        }*/
    }
}
```

### Error

```
* azurerm_virtual_machine.ap-service-discovery-server-0: autorest:DoErrorUnlessStatusCode 400 PUT https://management.azure.com/subscriptions/8b086f2c-3ade-4770-8825-d97490e3fe83/resourceGroups/core-dev/providers/Microsoft.Compute/virtualMachines/ap-service-discovery-server-0?api-version=2015-06-15 failed with 400 Bad Request
```

from charles proxy:

```
"properties": {
			"statusCode": "BadRequest",
			"serviceRequestId": "2ab50479-ef29-4b24-95fe-8fd8967831cd",
			"statusMessage": "{\"error\":{\"code\":\"InvalidParameter\",\"target\":\"osDisk.osType\",\"message\":\"Required parameter 'osDisk.osType' is missing (null).\"}}"
		},
```

### Expected Behavior
What should have happened?
The VM should have successfully been created with the specified OS disk. 

osType is not mentioned in the example for the virtual machine resource and no mapping seems to exist.

### Actual Behaviour
error 400 bad request - see charles output above.

Hi @stack72,

I built the provider successfully, but I couldn't get past plan - so I assume the fix is in the core?

I tried building the whole lot just using make - but got the following issue:

ok  	github.com/hashicorp/terraform	0.023s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-atlas	0.019s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-aws	0.022s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-azure	0.020s
?   	github.com/hashicorp/terraform/builtin/bins/provider-azurerm	[no test files]
?   	github.com/hashicorp/terraform/builtin/bins/provider-chef	[no test files]
ok  	github.com/hashicorp/terraform/builtin/bins/provider-clc	0.018s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-cloudflare	0.023s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-cloudstack	0.020s
?   	github.com/hashicorp/terraform/builtin/bins/provider-cobbler	[no test files]
ok  	github.com/hashicorp/terraform/builtin/bins/provider-consul	0.034s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-datadog	0.023s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-digitalocean	0.019s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-dme	0.019s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-dnsimple	0.020s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-docker	0.019s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-dyn	0.020s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-fastly	0.018s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-github	0.019s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-google	0.021s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-heroku	0.021s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-influxdb	0.020s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-librato	0.019s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-mailgun	0.021s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-mysql	0.021s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-null	0.018s
?   	github.com/hashicorp/terraform/builtin/bins/provider-openstack	[no test files]
?   	github.com/hashicorp/terraform/builtin/bins/provider-packet	[no test files]
ok  	github.com/hashicorp/terraform/builtin/bins/provider-postgresql	0.020s
ok  	github.com/hashicorp/terraform/builtin/bins/provider-powerdns	0.019s
?   	github.com/hashicorp/terraform/builtin/bins/provider-rundeck	[no test files]
ok  	github.com/hashicorp/terraform/builtin/bins/provider-softlayer	0.020s
?   	github.com/hashicorp/terraform/builtin/bins/provider-statuscake	[no test files]
?   	github.com/hashicorp/terraform/builtin/bins/provider-template	[no test files]
ok  	github.com/hashicorp/terraform/builtin/bins/provider-terraform	0.021s
?   	github.com/hashicorp/terraform/builtin/bins/provider-tls	[no test files]
?   	github.com/hashicorp/terraform/builtin/bins/provider-triton	[no test files]
?   	github.com/hashicorp/terraform/builtin/bins/provider-ultradns	[no test files]
?   	github.com/hashicorp/terraform/builtin/bins/provider-vcd	[no test files]
ok  	github.com/hashicorp/terraform/builtin/bins/provider-vsphere	0.028s
ok  	github.com/hashicorp/terraform/builtin/bins/provisioner-chef	0.020s
ok  	github.com/hashicorp/terraform/builtin/bins/provisioner-file	0.021s
ok  	github.com/hashicorp/terraform/builtin/bins/provisioner-local-exec	0.019s
ok  	github.com/hashicorp/terraform/builtin/bins/provisioner-remote-exec	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/atlas	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/aws	1.702s
ok  	github.com/hashicorp/terraform/builtin/providers/azure	0.032s
ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	0.027s
ok  	github.com/hashicorp/terraform/builtin/providers/chef	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/clc	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/cloudflare	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/cloudstack	0.022s
ok  	github.com/hashicorp/terraform/builtin/providers/cobbler	0.022s
ok  	github.com/hashicorp/terraform/builtin/providers/consul	0.020s
ok  	github.com/hashicorp/terraform/builtin/providers/datadog	0.020s
ok  	github.com/hashicorp/terraform/builtin/providers/digitalocean	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/dme	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/dnsimple	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/docker	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/dyn	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/fastly	0.020s
ok  	github.com/hashicorp/terraform/builtin/providers/github	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/google	0.034s
ok  	github.com/hashicorp/terraform/builtin/providers/heroku	0.022s
ok  	github.com/hashicorp/terraform/builtin/providers/influxdb	0.020s
ok  	github.com/hashicorp/terraform/builtin/providers/librato	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/mailgun	0.020s
ok  	github.com/hashicorp/terraform/builtin/providers/mysql	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/null	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/openstack	0.022s
ok  	github.com/hashicorp/terraform/builtin/providers/packet	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/postgresql	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/powerdns	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/rundeck	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/softlayer	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/statuscake	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/template	0.019s
ok  	github.com/hashicorp/terraform/builtin/providers/terraform	0.021s
ok  	github.com/hashicorp/terraform/builtin/providers/test	0.212s
ok  	github.com/hashicorp/terraform/builtin/providers/tls	0.021s
ok  	github.com/hashicorp/terraform/builtin/providers/triton	0.021s
ok  	github.com/hashicorp/terraform/builtin/providers/ultradns	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/vcd	0.018s
ok  	github.com/hashicorp/terraform/builtin/providers/vsphere	0.032s
ok  	github.com/hashicorp/terraform/builtin/provisioners/chef	0.033s
ok  	github.com/hashicorp/terraform/builtin/provisioners/file	0.018s
ok  	github.com/hashicorp/terraform/builtin/provisioners/local-exec	0.054s
ok  	github.com/hashicorp/terraform/builtin/provisioners/remote-exec	0.018s
ok  	github.com/hashicorp/terraform/command	1.446s
ok  	github.com/hashicorp/terraform/communicator	0.019s
ok  	github.com/hashicorp/terraform/communicator/remote	0.011s
ok  	github.com/hashicorp/terraform/communicator/ssh	0.056s
ok  	github.com/hashicorp/terraform/communicator/winrm	0.034s
ok  	github.com/hashicorp/terraform/config	0.077s
ok  	github.com/hashicorp/terraform/config/module	0.036s
ok  	github.com/hashicorp/terraform/dag	0.014s
ok  	github.com/hashicorp/terraform/digraph	0.011s
?   	github.com/hashicorp/terraform/dot	[no test files]
ok  	github.com/hashicorp/terraform/flatmap	0.012s
?   	github.com/hashicorp/terraform/helper/acctest	[no test files]
ok  	github.com/hashicorp/terraform/helper/config	0.018s
ok  	github.com/hashicorp/terraform/helper/diff	0.018s
ok  	github.com/hashicorp/terraform/helper/hashcode	0.009s
?   	github.com/hashicorp/terraform/helper/logging	[no test files]
ok  	github.com/hashicorp/terraform/helper/mutexkv	0.062s
ok  	github.com/hashicorp/terraform/helper/pathorcontents	0.014s
ok  	github.com/hashicorp/terraform/helper/resource	15.061s
ok  	github.com/hashicorp/terraform/helper/schema	0.023s
ok  	github.com/hashicorp/terraform/plugin	0.259s
ok  	github.com/hashicorp/terraform/rpc	0.038s
ok  	github.com/hashicorp/terraform/state	0.023s
2016/05/09 17:22:09 [DEBUG] PUT http://127.0.0.1:49441/api/v1/terraform/state/someuser/some-test-remote-state?access_token=sometoken&atlas_run_id=
2016/05/09 17:22:09 [DEBUG] Handling Atlas conflict response: Conflict
2016/05/09 17:22:09 [DEBUG] Atlas reported conflict, checking for equivalent states.
2016/05/09 17:22:09 [DEBUG] GET http://127.0.0.1:49441/api/v1/terraform/state/someuser/some-test-remote-state?access_token=sometoken&atlas_run_id=
2016/05/09 17:22:09 [DEBUG] States are equivalent, incrementing serial and retrying.
2016/05/09 17:22:09 [DEBUG] PUT http://127.0.0.1:49441/api/v1/terraform/state/someuser/some-test-remote-state?access_token=sometoken&atlas_run_id=
2016/05/09 17:22:09 [DEBUG] PUT http://127.0.0.1:49445/api/v1/terraform/state/someuser/some-test-remote-state?access_token=sometoken&atlas_run_id=
2016/05/09 17:22:09 [DEBUG] PUT http://127.0.0.1:49447/api/v1/terraform/state/someuser/some-test-remote-state?access_token=sometoken&atlas_run_id=
2016/05/09 17:22:09 [DEBUG] Handling Atlas conflict response: Conflict
2016/05/09 17:22:09 [DEBUG] Atlas reported conflict, checking for equivalent states.
2016/05/09 17:22:09 [DEBUG] GET http://127.0.0.1:49447/api/v1/terraform/state/someuser/some-test-remote-state?access_token=sometoken&atlas_run_id=
2016/05/09 17:22:09 [DEBUG] States are not equivalent, returning conflict.
2016/05/09 17:22:09 [DEBUG] PUT http://127.0.0.1:49450/api/v1/terraform/state/someuser/some-test-remote-state?access_token=sometoken&atlas_run_id=
--- FAIL: TestAtlasClient_UnresolvableConflict (0.05s)
	atlas_test.go:163: Timed out after 50ms, probably because retrying infinitely.
2016/05/09 17:22:09 [ERR] PUT http://127.0.0.1:49450/api/v1/terraform/state/someuser/some-test-remote-state?access_token=sometoken&atlas_run_id= request failed: Put http://127.0.0.1:49450/api/v1/terraform/state/someuser/some-test-remote-state?access_token=sometoken&atlas_run_id=: read tcp 127.0.0.1:49451->127.0.0.1:49450: read: connection reset by peer
2016/05/09 17:22:09 [DEBUG] PUT http://127.0.0.1:49450/api/v1/terraform/state/someuser/some-test-remote-state?access_token=sometoken&atlas_run_id=: retrying in 1s
2016/05/09 17:22:10 [WARN] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
FAIL
FAIL	github.com/hashicorp/terraform/state/remote	1.496s
ok  	github.com/hashicorp/terraform/terraform	0.971s
make: *** [test] Error 1

Is this a common error?

--- a/builtin/providers/azurerm/resource_arm_virtual_machine.go
+++ b/builtin/providers/azurerm/resource_arm_virtual_machine.go
@@ -118,6 +118,11 @@ func resourceArmVirtualMachine() *schema.Resource {
 				Required: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						"os_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
 						"name": &schema.Schema{
 							Type:     schema.TypeString,
 							Required: true,
@@ -126,6 +131,11 @@ func resourceArmVirtualMachine() *schema.Resource {
 						"vhd_uri": &schema.Schema{
 							Type:     schema.TypeString,
 							Required: true,
+						},
+
+						"image_uri": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
 						},
 
 						"caching": &schema.Schema{
@@ -1075,6 +1085,7 @@ func expandAzureRmVirtualMachineOsDisk(d *schema.ResourceData) (*compute.OSDisk,
 
 	name := disk["name"].(string)
 	vhdURI := disk["vhd_uri"].(string)
+	imageURI := disk["image_uri"].(string)
 	createOption := disk["create_option"].(string)
 
 	osDisk := &compute.OSDisk{
@@ -1083,6 +1094,22 @@ func expandAzureRmVirtualMachineOsDisk(d *schema.ResourceData) (*compute.OSDisk,
 			URI: &vhdURI,
 		},
 		CreateOption: compute.DiskCreateOptionTypes(createOption),
+	}
+
+	if v := disk["image_uri"].(string); v != "" {
+		osDisk.Image = &compute.VirtualHardDisk{
+			URI: &imageURI,
+		}
+	}
+
+	if v := disk["os_type"].(string); v != "" {
+		if v == "linux" {
+			osDisk.OsType = compute.Linux
+		} else if v == "windows" {
+			osDisk.OsType = compute.Windows
+		} else {
+			return nil, fmt.Errorf("[ERROR] os_type must be 'linux' or 'windows'")
+		}
 	}
 
 	if v := disk["caching"].(string); v != "" {

--- a/website/source/docs/providers/azurerm/r/virtual_machine.html.markdown
+++ b/website/source/docs/providers/azurerm/r/virtual_machine.html.markdown
@@ -137,6 +137,8 @@ For more information on the different example configurations, please check out t
 * `vhd_uri` - (Required) Specifies the vhd uri.
 * `create_option` - (Required) Specifies how the virtual machine should be created. Possible values are `attach` and `FromImage`.
 * `caching` - (Optional) Specifies the caching requirements.
+* `image_uri` - (Optional) Specifies the image_uri in the form publisherName:offer:skus:version.
+* `os_type` - (Optional) Specifies the operating system Type, valid values are windows, linux.
 
 `storage_data_disk` supports the following:
 

